lib/matplotlib/animation.py

```diff
@@ -337,6 +347,7 @@ def grab_frame(self, **savefig_kwargs):
         verbose.report('MovieWriter.grab_frame: Grabbing frame.',
                        level='debug')
         try:
+            self._adjust_frame_size()
```

Comment:
This only partially prevents the problem if the callbacks always set the same size.  Probably should cache the size (and dpi) in setup and then re-set them here.